AppTaskQueue/appscale/taskqueue/distributed_tq.py

```diff
@@ -632,6 +673,85 @@ def get_task_args(self, request):
                                   retry_parameters().max_doublings()
     return args
 
+  def get_target_from_queue(self, app_id, target_instance, target):
+    """ Gets the url for the target using the queue's target defined in the
+    configuration file.
+    
+    Args:
+      app_id: The application id, used to lookup module port.
+      target_instance: The instance ip to use for the new url.
+      target: A string containing the value of queue.target.
+    Returns:
+       A url as a string for the given target.
+    """
+    target_info = target.split('.')
+    return "http://{ip}:{port}".format(ip=target_instance,
+             port=self.get_module_port(app_id, target_info))
+
+  def get_target_from_host(self, app_id, target_instance, host):
+    """ Gets the url for the target using the Host header.
+    
+    Args:
+      app_id: The application id, used to lookup module port.
+      target_instance: The instance ip to use for the new url.
+      host: A string containing the value of the Task's host from target or
+        the HTTP_HOST (which would contain AppScale's login ip).
+        
+    Returns:
+      A url as a string for the given target or None if target contains the
+        AppScale login ip because the Task did not specify a target. If this
+        method returns None the target will be determined by the queue or use
+        the current running version and module.
+    """
+    if not TARGET_REGEX.match(host):
+      return None
```

Comment:
Unfortunately the taskqueue api side will either set Host to version.module or HTTP_HOST. if it is set to HTTP_HOST we should not raise an error because that means the task didn't specify a target.